storage/storage_src.go

```diff
@@ -302,9 +302,6 @@ func (s *storageImageSource) LayerInfosForCopy(ctx context.Context, instanceDige
 		if err != nil {
 			return nil, fmt.Errorf("reading layer %q in image %q: %w", layerID, s.image.ID, err)
 		}
-		if layer.UncompressedSize < 0 {
-			return nil, fmt.Errorf("uncompressed size for layer %q is unknown", layerID)
-		}
```

Comment:
FIXME

Reply:
Thanks for the comment. According to the docs, unknown BlobInfo.Size seems to be allowed: https://github.com/containers/image/blob/6caf212454493ef1979393c398ac705aebb393a6/types/types.go#L124

Reply:
More importantly, the documentation of `LayerInfosForCopy` does allow the size to be not returned.

But that seemed surprising to me — hence the FIXME — because in modern manifest formats, the size is mandatory. So that FIXME was a note to self… and it turns out that all of the blob copy path can deal with `LayerInfosForCopy` not carrying a size fine, except for the code path where the layer is using a remote URL and we don’t copy anything.

This function never includes an `URLs` value (discarding one if it were present in the original manifest), so we are fine.